apis/Google.Cloud.Firestore.Data/Google.Cloud.Firestore.Data/FirestoreDb.cs

```diff
@@ -0,0 +1,148 @@
+        public string ProjectId { get; }
+
+        /// <summary>
+        /// The database ID associated with this database.
+        /// </summary>
+        public string DatabaseId { get; }
+
+        /// <summary>
+        /// The resource name of the database, in the form "projects/{project_id}/databases/{database_id}".
+        /// </summary>
+        internal string RootPath { get; }
+
+        /// <summary>
+        /// The root of the documents, in the form "projects/{project_id}/databases/{database_id}/documents".
+        /// </summary>
+        internal string DocumentsPath { get; }
+
+        private FirestoreDb(string projectId, string databaseId, FirestoreClient client)
+        {
+            ProjectId = GaxPreconditions.CheckNotNull(projectId, nameof(projectId));
+            DatabaseId = GaxPreconditions.CheckNotNull(databaseId, nameof(databaseId));
+            Client = GaxPreconditions.CheckNotNull(client, nameof(client));
+            RootPath = $"projects/{ProjectId}/databases/{DatabaseId}";
+            DocumentsPath = $"{RootPath}/documents";
+        }
+
+        /// <summary>
+        /// Creates an instance for the specified project and database, using the specified <see cref="FirestoreClient"/>
+        /// for RPC operations.
+        /// </summary>
+        /// <param name="projectId">The ID of the Google Cloud Platform project that contains the database. May be null, in which case
+        /// the project will be automatically detected if possible.</param>
+        /// <param name="databaseId">The ID of the database within the project. May be null, in which case the default database will be used.</param>
+        /// <param name="client">The client to use for RPC operations. May be null, in which case a client will be created with default credentials.</param>
+        /// <returns>A new instance.</returns>
+        public static FirestoreDb Create(string projectId = null, string databaseId = null, FirestoreClient client = null) =>
+            new FirestoreDb(
+                projectId ?? Platform.Instance().ProjectId,
+                databaseId ?? DefaultDatabaseId,
+                client ?? FirestoreClient.Create());
+
+        /// <summary>
+        /// Asynchronously creates an instance for the specified project and database, using the specified <see cref="FirestoreClient"/>
+        /// for RPC operations.
+        /// </summary>
+        /// <param name="projectId">The ID of the Google Cloud Platform project that contains the database. May be null, in which case
+        /// the project will be automatically detected if possible.</param>
+        /// <param name="databaseId">The ID of the database within the project. May be null, in which case the default database will be used.</param>
+        /// <param name="client">The client to use for RPC operations. May be null, in which case a client will be created with default credentials.</param>
+        /// <returns>A new instance.</returns>
```

Comment:
Should this be updated to mention the task?

Reply:
Done.